src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -115,6 +118,92 @@ class RemoteMachineTrainingService implements TrainingService {
         }
         this.log.info('Remote machine training service exit.');
     }
+    
+    public async setTrialSSHConnection(trial: RemoteMachineTrialJobDetail): Promise<void> {
+        const deferred: Deferred<void> = new Deferred<void>();
+        if(!trial.rmMeta) {
+            throw new Error(`rmMeta not set in trial ${trial.id}`);
+        }
+        let sshClientManager: SSHClientManager | undefined = this.machineSSHClientMap.get(trial.rmMeta);
+        if(!sshClientManager) {
+            throw new Error(`remoteSSHClient not initialized`);
+        }
+        let sshClient: Client | undefined = sshClientManager.getAvailableSSHClient();
+        if(!sshClient) {
+            //create a new connection
+            await this.createSSHClient(trial);
+        }else {
+            this.trialSSHClientMap.set(trial.id, sshClient);
+        }
+
+        deferred.resolve();
+        return deferred.promise;
+    }
+
+    /**
+     * Create a new ssh connection client by trial job id
+     */
+    public async createSSHClient(trial: RemoteMachineTrialJobDetail): Promise<void> {
```

Comment:
The naming of this function is confused.

Reply:
You should also consider put SSH client initialization into SSH connection manager.

Reply:
1.changed `createSSHClient` to `scheduleTrialSSHClient`  

2.move create new ssh client logic to SSHClientManager